hetmech/matrix.py

```diff
@@ -19,7 +19,7 @@ def get_node_to_position(graph, metanode):
 
 
 def metaedge_to_adjacency_matrix(graph, metaedge, dtype=numpy.bool_,
-                                 matrix_type=numpy.array):
+                                 auto=False, matrix_type=numpy.ndarray):
```

Comment:
Can we change `auto` to `sparse_threshold` where `if density > sparse_threshold: convert_to_dense`? Then you can set `sparse_threshold` to 0 to disable the conversion.